### PR TITLE
[CBRD-24327] When checking a synonym before checking a class in the semantic check, if the synonym does not exist, the error is cleared.

### PR DESCRIPTION
**Pull request by youngjinj**
http://jira.cubrid.org/browse/CBRD-24327

### When checking a synonym before checking a class in the semantic check, if the synonym does not exist, the error is cleared.
- The db_find_synonym () and sm_find_synonym () functions set an ER_SYNONYM_NOT_EXIST error when a synonym does not exist.

### CODE REVIEW

File: src/query/execute_schema.c [hunk 1]
```diff
@@ -2623,11 +2623,20 @@ do_rename (PARSER_CONTEXT * parser, PT_NODE * statement)
...
 	  PT_ERRORmf (parser, statement, MSGCAT_SET_PARSER_SEMANTIC, MSGCAT_SEMANTIC_CLASS_DOES_NOT_EXIST, old_name);
 	  goto error_exit;
 	}
+      else
+	{
+	  /* db_find_synonym () == NULL */
+	  if (er_errid () == ER_SYNONYM_NOT_EXIST)
+	    {
+	      er_clear ();
+	    }
```
Comment (ctshim) [inline src/query/execute_schema.c]: db_find_synonym() returned NULL.
If it is due to an error other than ER_SYNONYM_NOT_EXIST, shouldn't it be handled as an error without ignoring it?

 ```
     if (er_errid () == ER_SYNONYM_NOT_EXIST)
	    {
	      er_clear ();
	    }
     else
	    {
	      // PT_ERRORmf( ... )
	      goto error_exit;
	    }    
```

File: src/parser/semantic_check.c [hunk 2]
```diff
@@ -9486,11 +9541,21 @@ pt_check_truncate (PARSER_CONTEXT * parser, PT_NODE * node)
       if ((name = temp->info.spec.entity_name) != NULL && name->node_type == PT_NAME
 	  && (cls_nam = name->info.name.original) != NULL)
 	{
+	  /* We cannot change the schema of a class by using synonym names. */
```
Comment (ctshim) [inline src/parser/semantic_check.c:9676]: Can't truncate with synonym names?

Comment (ctshim) [reply src/parser/semantic_check.c:9676]: I got an explanation from the author. The question has been resolved.

**Review by hyunikn — APPROVED** [review]

**Review by hgryoo — APPROVED** [review]

File: src/query/execute_schema.c (hunk 1 above)
Comment (youngjinj) [reply src/query/execute_schema.c]: Added error handling for cases other than ER_SYNONYM_NOT_EXIST errors.

**Review by beyondykk9 — APPROVED** [review]

**Review by ctshim — APPROVED** [review]

**Review by shparkcubrid — APPROVED** [review]

**Pull request merged**